### PR TITLE
Benji/ios 423 update people vc

### PR DESCRIPTION
**Pull request by Benjamindodgson**
https://user-images.githubusercontent.com/5214531/151080308-0bb80f62-62ea-4be6-919e-247e5decba7f.mov

Still having some text reuse issues but other than that, its way better.

### CODE REVIEW

File: Shared/UI/Views/Gradients/GradientView.swift [hunk 1]
```diff
@@ -44,7 +44,7 @@ class GradientLayer: CAGradientLayer {
     }
 }
 
-class GradientView: PassThroughView {
+class GradientView: BaseView {
```
Comment (Benjamindodgson) [inline Shared/UI/Views/Gradients/GradientView.swift:47]: lol whoops

File: Shared/UI/Views/ThemeLabel.swift [hunk 2]
```diff
@@ -101,22 +99,26 @@ class ThemeLabel: UILabel {
...
 
     func add(attributes: [NSAttributedString.Key : Any], to text: String) {
-        guard let existingText = self.attributedText, let range = existingText.string.range(of: text) else {
-            return
-        }
+        guard let existingText = self.attributedText else { return }
         
+        let lowercased = existingText.string.lowercased()
+        
+        guard let range = lowercased.range(of: text.lowercased()) else { return }
```
Comment (Benjamindodgson) [inline Shared/UI/Views/ThemeLabel.swift:109]: Ensures casing isn't an issue.

File: Shared/UI/Views/ThemeLabel.swift [hunk 3]
```diff
@@ -101,22 +99,26 @@ class ThemeLabel: UILabel {
...
+                
         let attributedString = NSMutableAttributedString(existingText)
         attributedString.addAttributes(attributes, range: range.nsRange(text))
         super.attributedText = attributedString
     }
 
     private func setTextWithAttributes(_ newText: String) {
         let string = self.stringCasing.format(string: newText)
+        
+        let attributes = self.attributedText?.attributes(at: 0, effectiveRange: nil) ?? self.defaultAttributes
```
Comment (Benjamindodgson) [inline Shared/UI/Views/ThemeLabel.swift:119]: If we already have attributes, use them.

File: iOS/Collection View Management /CollectionView.swift [hunk 4]
```diff
@@ -24,15 +25,16 @@ class CollectionView: UICollectionView {
     }
 
     func initializeViews() {
-        let backView = BaseView()
-        backView.set(backgroundColor: .clear)
-        self.backgroundView = backView
-        backView.addSubview(self.animationView)
+        self.backView.set(backgroundColor: .clear)
```
Comment (Benjamindodgson) [inline iOS/Collection View Management /CollectionView.swift:28]: This fixes issue with loading spinner sometimes not centering.

File: iOS/Collection View Management /CollectionViewManagerCell.swift [hunk 5]
```diff
@@ -45,7 +46,9 @@ class CollectionViewManagerCell: UICollectionViewListCell {
 
     }
 
-    func update(isSelected: Bool) {}
+    func update(isSelected: Bool) {
+        guard isSelected != self.isSelected else { return }
```
Comment (Benjamindodgson) [inline iOS/Collection View Management /CollectionViewManagerCell.swift:50]: Initial load causes this to get called for every visible cell.

File: Shared/UI/Views/Gradients/GradientView.swift (hunk 1 above)
Comment (MartinJibber) [reply Shared/UI/Views/Gradients/GradientView.swift:47]: lol

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**